src/responses.js

```diff
@@ -0,0 +1,179 @@
+const { fixArraySchema, hasFileRef, getRefFromInclude, searchDataStructure } = require('./util')
+const escapeJSONPointer = require('./escape_json_pointer')
+const isEqual = require('lodash.isequal')
+const http = require('http')
+
+const parseResponseSchema = (schema, openApi3) => {
+    if (!schema) return
+    if (openApi3 && hasFileRef(schema)){
+        return getRefFromInclude(schema) 
+    }
+    try {
+        const result = JSON.parse(schema);
+        delete result['$schema'];
+        fixArraySchema(result); // work around for Swagger UI / Editor
+        return result
+    } catch (e) { }
+}
+
+const parseResponseBody = (body, header, openApi3) => {
+    if (openApi3 && hasFileRef(body)){
+        return getRefFromInclude(body)
+    }
+    if (!header.value.match(/application\/.*json/)) {
+        return body
+    } 
+    try {
+        return JSON.parse(body);
+    } catch (e) { }
+}
+
+const getResponseSchema = (response, options) => {
+    const { preferReference, openApi3 } = options
+    const componentsPath = openApi3 ? '#/components/schemas' : '#/definitions/'
+    if (preferReference) { // MSON then schema
+        const inputSchema = searchDataStructure(response.content, openApi3); // Attributes in response
+        if (inputSchema) {
+            return inputSchema
+        } else if (response.reference) {
+            return {
+                '$ref': componentsPath + escapeJSONPointer(response.reference.id + 'Model')
+            };
+        } else if (response.schema) {
+            return parseResponseSchema(response.schema, openApi3)
+        }
+    } else { // schema then MSON
+        if (response.schema) {
+            return parseResponseSchema(response.schema, openApi3)
+        }
+       
+        const inputSchema = searchDataStructure(response.content, openApi3); // Attributes in response
+        if (inputSchema) return inputSchema;
+        if (response.reference) {
+            return {
+                '$ref': componentsPath + escapeJSONPointer(response.reference.id + 'Model')
+            };
+        }
+    }
+}
+
+const setResponseSchema = (responses, response, schema, openApi3) => {
```

Comment:
Instead of explicitly passing in `openapi3`, let's use the `options` variable or `context` to derive the state